src/Eyes/Db/PostgreSQL/Pdo.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace BeholderWebClient\Eyes\Db\PostgreSQL;
+
+use PDO as PHPDO;
+use Exception;
+use BeholderWebClient\Eyes\Db\DbStatus as Status;
+
+class Pdo  extends AbstractAdapter {
+
+  protected $pdo;
+  protected $ran = false;
+
+  public function checkRequirement(){
+
+    if(!class_exists("PDO"))
+      throw new Exception(parent::PREFIX_RQ_FAIL . 'PDO driver not found', Status::INTERNAL_SERVER_ERROR_NUMBER);
+
+  }
+
+  public function testConn(){
+
+    try {
+
+      $dns = "pgsql:host={$this->conf['host']};dbname={$this->conf['dbname']};port={$this->conf['port']};";
+      $this->pdo = new PHPDO($dns, $this->conf['user'] , $this->conf['password'],
+      [PHPDO::ATTR_ERRMODE => PHPDO::ERRMODE_EXCEPTION]);
+
+    }catch(Exception $ex){
+      parent::throwCouldNotConnect($ex->getMessage());
+    }
+
+  }
+
+  public function testQuery(){
+
+      if(isset($this->conf['query']['create']))
+        $this->execCreatequery('create');
+
+      if(isset($this->conf['query']['insert']))
+        $this->execquery('insert', Status::QUERY_INSERT_FAIL_NUMBER, Status::QUERY_INSERT_FAIL);
+
+      if(isset($this->conf['query']['update']))
+        $this->execquery('update', Status::QUERY_UPDATE_FAIL_NUMBER, Status::QUERY_UPDATE_FAIL);
+
+      if(isset($this->conf['query']['select']))
+        $this->selectquery();
+
+      if(isset($this->conf['query']['delete']))
+        $this->execquery('delete', Status::QUERY_DELETE_FAIL_NUMBER, Status::QUERY_DELETE_FAIL);
+
+      if(isset($this->conf['query']['drop']))
+        $this->execDropquery('drop');
+
+      if(!$this->ran)
+        parent::throwQueryBadFormated();
+
+  }
+
+  protected function execquery($type, $errNo, $errMessage){
+
+    $this->ran = true;
```

Comment:
This is supose to be true, it start at line 12 as false, and if it "ran" then ran is true.